src/Illuminate/Database/Eloquent/InvalidIdFormatException.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Illuminate\Database\Eloquent;
+
+use Illuminate\Support\Arr;
+
+/**
+ * @template TModel of \Illuminate\Database\Eloquent\Model
+ */
+class InvalidIdFormatException extends \RuntimeException
+{
+    /**
+     * Name of the affected Eloquent model.
+     *
+     * @var class-string<TModel>
+     */
+    protected $model;
+
+    /**
+     * The affected model IDs.
+     *
+     * @var array<int, int|string>
+     */
+    protected $ids;
+
+    /**
+     * Set the affected Eloquent model and instance id.
+     *
+     * @param  class-string<TModel>  $model
+     * @param  array<int, int|string>|int|string  $ids
+     * @param  string|null  $field
+     * @return $this
+     */
+    public function setModel($model, $ids = [], ?string $field = null)
+    {
+        $this->model = $model;
+        $this->ids = Arr::wrap($ids);
+
+        $this->message = 'Invalid key';
+        if ($field !== null) {
+            $this->message .= " [{$field}]";
+        }
+
+        $this->message .= " for model [{$model}]";
+
+        if (count($this->ids) > 0) {
```

Comment:
This can be simplified by omitting the function call and comparing it to an empty scalar value:
```suggestion
        if ($this->ids !== []) {
```